data/transactions/logic/eval_test.go

```diff
@@ -2036,6 +2056,85 @@ global ZeroAddress
 	require.True(t, pass)
 }
 
+func TestTxnas(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	t.Parallel()
+
+	source := `int 1
+txnas Accounts
+int 0
+txnas ApplicationArgs
+==
+`
+	ops := testProg(t, source, AssemblerMaxVersion)
+	var txn transactions.SignedTxn
+	txn.Txn.Accounts = make([]basics.Address, 1)
+	txn.Txn.Accounts[0] = txn.Txn.Sender
+	txn.Txn.ApplicationArgs = make([][]byte, 1)
+	txn.Txn.ApplicationArgs[0] = []byte(protocol.PaymentTx)
+	txgroup := make([]transactions.SignedTxn, 1)
+	txgroup[0] = txn
+	ep := defaultEvalParams(nil, &txn)
+	ep.TxnGroup = txgroup
+	_, err := Eval(ops.Program, ep)
```

Comment:
I think you're just checking that the programs runs, not that it passes? It would be a stronger test to instead check that a simple use of `txnas` passes. I'd suggest something like the following, and you could adapt it for `gtxnas` and `gtxnsas` too.

```
int 0
txnas Accounts
txna Accounts 0
==
int 0
txnas ApplicationArgs
txna ApplicationArgs 0
==
&&
int 1
txnas ApplicationArgs
txna ApplicationArgs 1
==
&&
return
```

Reply:
Good point, I will check these with e.g. `testAccepts(t, "int 0; txnas Accounts; txna Accounts 0; ==", 5)`

Reply:
I merged what existed, I'll add these.

Reply:
I concluded these were fine.  The following tests checked `pass`, and they were grabbing the accounts and comparing them to app args.